src/contrib/mlir/lowerer.cpp

```diff
@@ -534,91 +534,77 @@ namespace
     {
         auto concat = cast<NGConcatOp>(op);
         auto loc = concat.getLoc();
-
         ScopedContext scope(rewriter, loc);
 
         // Create Value for result, and extract type info.
         Value* result = m_pass.buildOutputDefs(op, rewriter)[0];
         NGRAPH_CHECK(result, "Unexpected null result in ConcatOp");
+        auto resultTy = result->getType().dyn_cast<MemRefType>();
+        NGRAPH_CHECK(resultTy, "Unexpected non-memref result type");
```

Comment:
Sorry, my bad here. I missed the `dyn_cast` before. You can use `cast` instead of `dyn_cast` and remove the `NGRAPH_CHECK` since `cast` will assert if the casting doesn't succeed. That's the way it's usually done in LLVM.